packages/ccd-js-gen/README.md

```diff
@@ -0,0 +1,484 @@
+# Smart Contract Client Generator <!-- omit in toc -->
+
+[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.0-4baaaa.svg)](https://github.com/Concordium/.github/blob/main/.github/CODE_OF_CONDUCT.md)
+
+Generate TypeScript/JavaScript code for interating with smart contracts and modules on the Concordium blockchain.
+
+- Functions for dry-running and calling entrypoints of the smart contract.
+- Functions for constructing parameters.
+- Parsing logged events, return values and error messages.
+- Structured types for parameter, logged events, return values and error messages.
+
+The code is generated from deployable smart contract modules, meaning it can be done with any smart contract available locally and any smart contract deployed on chain.
+
+### Example usage of a generated client
+
+```typescript
+import * as SDK from "@concordium/node-sdk"; // or use @concordium/web-sdk.
+import * as MyContract from "./generated/my-contract.js"; // Code generated from a smart contract module.
+
+const grpcClient = // Setup gRPC client code here ...
+const contractAddress = SDK.ContractAddress.create(42) // Address of the smart contract.
+const signer = // Keys for signing an account transaction.
+const sender = // AccountAddress
```

Comment:
Wouldn't it be better to show some valid inputs here?

Reply:
I don't think so, I do this to reduce the noise in the example, such that the interesting parts are more visible.
The point of this example is more of a showcase to help the reader understand what this tool is providing.

Reply:
But it's an ad-hoc mixture of things that are defined and things that are not. Like the contract address is defined, but the sender account is not, nor is the grpc-client, and all of these look very much the same complexity wise to me.